### PR TITLE
Solved #319 Improve Mentor Page and optimize for Mobile view. [GSSoC'23]

### PR DESCRIPTION
**Pull request by vishnu9865**
## Related Issue
Improve Mentor Page and optimize for Mobile view.
Based on proposed Figma File: [Figma File Link](https://www.figma.com/file/mcGBmU9AwU7KQg7NrH00cO/Grabtern-Mentor-Page?type=design&node-id=0%3A1&t=gSKXY7BmfhRvOV3K-1)

Closes: #319

## Description of Changes
Changes made to `[mentorUsername].page.js`:
 - Separated the Page into 2 sections.
 - Right section is the Mentor's Profile and the Left section is the Mentor's Sessions offered.
 - Also remade both Session Confirmation and Share Mentor Page Modal to be responsive and on theme with the Mentor Page.
 - Commented before every individual component to explain what it does for more readability.
 - Preserved the overall functions and logic as the previous Mentor Page (did not modify handleclick, useEffect, etc..).

## Checklist:

- [X] My code adheres to the established style guidelines of this project.
- [X] I have conducted a self-review of my code.
- [X] I have included comments in areas that may be difficult to understand.
- [X] I have made corresponding updates to the project documentation.
- [X] My changes have not introduced any new warnings.

## Screenshots

|           Original           |           Updated           |
| :--------------------------: | :-------------------------: |
| image | image |

### Share Page Modal
image
### Confirm Session Modal
image

@anmode Sir,
Please do feel free to ask for any changes if required.
Thank you for taking your time for this PR.
Regards,
Jeyavishnu @vishnu9865

### CODE REVIEW

**Review by anmode — APPROVED** [review]
Impressive! @vishnu9865

**Comment by anmode** [comment]
Screenshot 2023-06-14 at 6 42 53 PM
Just a question if this a component...why button changed it's position?

File: pages/[mentorUsername].page.js [hunk 1]
```diff
@@ -120,264 +125,188 @@ function Index({ mentorDetail }) {
...
+                <div className="tw-flex tw-flex-col">
+                  {/* Session card title */}
+                  <h1 className="tw-text-[36px] tw-font-semibold tw-text-[#4338CA]">{session.sessionName}</h1>
+                  {/* Session card description */}
+                  <h3 className="tw-text-[16px] tw-font-semibold tw-text-[#4338CA]/80">{session.sessionDescription}</h3>
+                  {/* Session card duration */}
+                  <p className="tw-flex tw-flex-row tw-items-center tw-gap-3 tw-text-[16px] tw-whitespace-nowrap tw-font-medium tw-text-black/50"><BsClock /> {session.sessionMeetingDuration} mins</p>
+                  {/* Session card Charge */}
+                  <p className="tw-flex tw-flex-row tw-items-center tw-gap-3 tw-text-[16px] tw-whitespace-nowrap tw-font-medium tw-text-black/50"><BsCurrencyRupee /> {session.priceSession} mins</p>
+                  {/* Session Book Button */}
```
Comment (anmode) [inline pages/[mentorUsername].page.js:219]: Can I have this as a component? Actually using a modular structure is very benificial. We will then pass on data. into these card

**Review by anmode — COMMENTED** [review]
Let's break the file into components now!.. You can make a folder `mentorProfile` in the components folder and include all component code there and can use components. 
Thanks. Btw page is looking damn awesome

**Comment by vishnu9865** [comment]
> Just a question if this a component...why button changed it's position?

Sir I will look into it now.

File: pages/[mentorUsername].page.js (hunk 1 above)
Comment (vishnu9865) [reply pages/[mentorUsername].page.js]: https://github.com/anmode/grabtern-frontend/pull/331/commits/2ffc694eb320cbc9a7467a1cdc18572419893d7d
@anmode Sir,
Done 🚀🚀👍👍

File: pages/[mentorUsername].page.js [hunk 2]
```diff
@@ -120,264 +125,188 @@ function Index({ mentorDetail }) {
...
+              <div className="tw-flex tw-flex-row tw-items-center tw-justify-end tw-gap-8">
+                <button 
+                  className="tw-bg-gray-500 tw-px-6 tw-py-3 tw-text-white tw-rounded-2xl"
+                  onClick={() => setModalPopup(false)}
+                >
+                  Cancel
+                </button>
+                {/* confirm button */}
+                <button 
+                  className="tw-bg-[#4338CA] tw-px-6 tw-py-3 tw-text-white tw-rounded-2xl hover:tw-bg-[#322995] active:tw-outline active:tw-outline-2 active:tw-outline-[#4338CA]/80 tw-transition-all"
```
[the remark this answers is not in the document]
Comment (vishnu9865) [reply pages/[mentorUsername].page.js]: https://github.com/anmode/grabtern-frontend/pull/331/commits/2ffc694eb320cbc9a7467a1cdc18572419893d7d
@anmode Sir,
Done 🚀🚀👍👍

**Review by anmode — APPROVED** [review]
Great Well done @vishnu9865!! Thanks a lot

**Comment by anmode** [comment]
@vishnu9865. Here is the offer- If you can learn Astro...would like to join us?

**Comment by vishnu9865** [comment]
> @vishnu9865. Here is the offer- If you can learn Astro...would like to join us?

@anmode Sir, I couldn't be more happy to say yes. I happily would learn astro for this wonderful project and team. This means so much to me. 🚀👍☺️☺️

**Comment by anmode** [comment]
> > @vishnu9865. Here is the offer- If you can learn Astro...would like to join us?
> 
> @anmode Sir, I couldn't be more happy to say yes. I happily would learn astro for this wonderful project and team. This means so much to me. 🚀👍☺️☺️

That's cool! @vishnu9865. Please ping me on discord. You can find me in project channel. 
To get into project channel- 
1. Go to discord server Gssoc
2. Select grabtern from #self-roles
3. you will find a private channel
4. Ping me

**Pull request merged**